app/src/components/collectionStructure/collectionStructure.tsx

```diff
@@ -1,258 +1,380 @@
-import { useState, forwardRef } from "react";
-import { Box, Button, Collapse } from "@chakra-ui/react";
+import { useState, useEffect, useRef } from "react";
+import { Box } from "@chakra-ui/react";
 import {
   Flex,
-  Icon,
-  Text,
   Heading,
-  Tooltip,
+  SkeletonLoader,
 } from "@nypl/design-system-react-components";
-import { mockCollectionChildrenResponse } from "__tests__/__mocks__/data/mockCollectionStructure";
-import { useScrollIntoViewIfNeeded } from "@/src/hooks/useScrollIntoViewIfNeeded";
+import { SearchManager } from "@/src/utils/searchManager";
 import { headerBreakpoints } from "@/src/utils/breakpoints";
+import AccordionTree from "./accordionTree";
+import { CARDS_PER_PAGE } from "@/src/config/constants";
 
-export interface CollectionChildProps {
+export interface OpenStateItem {
   title: string;
-  itemCount: string;
-  children?: CollectionChildProps[];
-  hasChildren: boolean;
   uuid: string;
-}
-
-interface OpenStateItem {
-  title: string;
   level: number;
+  itemCount: string | number;
   isOpen: boolean;
+  hasSubContainers: boolean;
+  children?: OpenStateItem[];
 }
 
-const ButtonText = ({ title, hasChildren, level }) => {
-  let text = (
-    <Text
-      flex="1"
-      marginBottom="0"
-      whiteSpace="nowrap"
-      overflow="hidden"
-      textOverflow="ellipsis"
-      paddingLeft={hasChildren ? "s" : "28px"}
-      fontWeight="500"
-    >
-      {title}
-    </Text>
-  );
-  let truncationLength = 30 - level * 7;
-  if (title.length > truncationLength) {
-    return (
-      <Tooltip zIndex="1000" content={text}>
-        {text}
-      </Tooltip>
-    );
+interface CollectionStructureProps {
+  uuid: string;
+  updateURL: (queryString: string) => Promise<void>;
+  searchManager: SearchManager;
+}
+
+const fetchChildren = async (uuid: string): Promise<OpenStateItem[]> => {
+  const allChildren: OpenStateItem[] = [];
+  let page = 1;
+  while (true) {
+    const res = await fetch(`/api/collectionchildren/${uuid}?page=${page}`);
+    if (!res.ok) throw new Error("Failed to fetch children");
```

Comment:
If this error is thrown, the component does not render and the results span full width.
<img width="546" alt="Screenshot 2025-05-09 at 6 16 55 PM" src="https://github.com/user-attachments/assets/48466480-71a3-466d-a8a1-97f7214c5390" />

Just curious, when was this decided? I don't necessarily disagree with this but, again, mostly curious. Given more time, I think it'd be nice to keep rendering the sidebar with an error message and a "try again" button.